aries_vcx/src/protocols/issuance/issuer/state_machine.rs

```diff
@@ -557,9 +557,15 @@ async fn _create_credential(
         ));
     };
     let request = &request.requests_attach.content()?;
-    let cred_data = encode_attributes(cred_data)?;
+    let cred_attrs = CredentialAttribute::new(cred_data).encode()?;
```

Comment:
Note similar to my comment around tests, this also seems to be bit of type mismatch here, as `CredentialAttribute` would represent just a single attribute

Reply:
I pushed another commit to address my mistake. I added a `CredentialAttributes::encode_all` API to encode all attributes in a credential. 

`CredentialAttributes` stores the attributes in an `attributes` table. However, using a vector will be better IMHO, but changing this breaks the tests, so I left it as it's now.